### PR TITLE
Feature/wizard internal services

### PR DESCRIPTION
**Pull request by annashamray**
Wizard config, step 2
Issue - #518
depends on https://github.com/open-zaak/open-zaak/pull/515

### CODE REVIEW

File: src/openzaak/config/migrations/0004_internalservices_initial.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    for component, api_type in COMPONENT_MAPPING.items():
+        InternalService.objects.create(api_type=api_type)
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("config", "0003_internalservice"),
+    ]
+
+    operations = [migrations.RunPython(initial)]
```
Comment (sergei-maertens) [inline src/openzaak/config/migrations/0004_internalservices_initial.py]:
```suggestion
    operations = [migrations.RunPython(initial, migrations.RunPython.noop)]
```

File: src/openzaak/config/models.py [hunk 2]
```diff
@@ -23,3 +26,33 @@ class Meta:
...
+
+
+class InternalService(models.Model):
+    api_type = models.CharField(
+        _("API type"), max_length=50, choices=ComponentTypes.choices, unique=True
+    )
+    enabled = models.BooleanField(
+        _("enabled"),
+        default=True,
+        help_text=_("Boolean indicates if internal API is enabled"),
```
Comment (sergei-maertens) [inline src/openzaak/config/models.py]:
```suggestion
        help_text=_("Indicates if the API is enabled in Open Zaak."),
```

File: src/openzaak/config/models.py [hunk 3]
```diff
@@ -23,3 +26,33 @@ class Meta:
...
+    )
+    nlx = models.BooleanField(
+        _("nlx"),
+        default=True,
+        help_text=_("Boolean indicated if the service is available via NLX inway"),
```
Comment (sergei-maertens) [inline src/openzaak/config/models.py]:
```suggestion
        help_text=_("Indicates if the service is to be provided over NLX."),
```

File: src/openzaak/config/models.py [hunk 4]
```diff
@@ -23,3 +26,33 @@ class Meta:
...
+    )
+
+    class Meta:
+        verbose_name = _("Internal service")
+        verbose_name_plural = _("Internal services")
+
+    def __str__(self):
+        return self.api_type
```
Comment (sergei-maertens) [inline src/openzaak/config/models.py]:
```suggestion
        return self.get_api_type_display()
```

File: src/openzaak/config/models.py [hunk 5]
```diff
@@ -23,3 +26,33 @@ class Meta:
...
+
+    @property
+    def component(self) -> str:
+        for component, api_type in COMPONENT_MAPPING.items():
+            if api_type == self.api_type:
+                return component
+        return ""
```
Comment (sergei-maertens) [inline src/openzaak/config/models.py]:
```suggestion
        raise ValueError(f"Unknown component for api_type '{api_type}'")
```

File: src/openzaak/config/templates/config/config_detail.html [hunk 6]
```diff
@@ -9,31 +9,64 @@ <h1 class="display-4">Service configuratie</h1>
...
+    <section>
+        <div class="text-center">
+            <h3>Open Zaak services</h3>
+        </div>
+
+        <table class="table table-striped">
+            <thead>
+                <tr>
+                    <th scope="col">Service</th>
+                    <th scope="col" class="text-center">Enabled</th>
```
Comment (sergei-maertens) [inline src/openzaak/config/templates/config/config_detail.html]: This needs some sort of info icon with a popover with additional information explaining the meaning of this setting.

File: src/openzaak/config/templates/config/config_detail.html [hunk 7]
```diff
@@ -9,31 +9,64 @@ <h1 class="display-4">Service configuratie</h1>
...
+                    <th scope="col" class="text-center">NLX</th>
```
Comment (sergei-maertens) [inline src/openzaak/config/templates/config/config_detail.html]: This needs some sort of info icon with a popover with additional information explaining the meaning of this setting.

File: src/openzaak/config/templates/config/config_internal.html [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    <form method="post" enctype="multipart/form-data">
+        {% csrf_token %}
+        {% load_widgets form='snippets/forms.html' %}
+
+        {{ formset.management_form }}
+        {{ formset.non_form_errors }}
+
+        <div class="form-group row">
+            <div class="col"><strong>Service</strong></div>
+            <div class="col text-center"><strong>Enabled?</strong></div>
```
Comment (sergei-maertens) [inline src/openzaak/config/templates/config/config_internal.html]: This needs some sort of info icon with a popover with additional information explaining the meaning of this setting.

File: src/openzaak/config/templates/config/config_internal.html [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+            <div class="col text-center"><strong>NLX?</strong></div>
```
Comment (sergei-maertens) [inline src/openzaak/config/templates/config/config_internal.html]: This needs some sort of info icon with a popover with additional information explaining the meaning of this setting.

File: src/openzaak/templates/snippets/forms.html [hunk 10]
```diff
@@ -54,3 +54,18 @@
     </div>
 </div>
 {% endblock %}
+
+
+{% block Toggle %}
+<div class="custom-control custom-switch">
+  <input name="{{ html_name }}" id="{{ id }}" type="checkbox" class="custom-control-input" {{ form_field.value|yesno:'checked,' }}>
```
Comment (sergei-maertens) [inline src/openzaak/templates/snippets/forms.html]:
```suggestion
  <input name="{{ html_name }}" id="{{ id }}" type="checkbox" class="custom-control-input" {{ raw_value|yesno:'checked,' }}>
```
This is the typed value in sniplates, I think this should work.

File: src/openzaak/templates/snippets/forms.html [hunk 11]
```diff
@@ -54,3 +54,18 @@
...
+  <label for="{{ form_field.id_for_label }}" class="custom-control-label"/>
+</div>
+{% endblock %}
+
+{% block ToggleDisabled %}
+<div class="custom-control custom-switch">
+  <input name="{{ html_name }}" id="{{ id }}" type="checkbox" class="custom-control-input" {{ form_field.value|yesno:'checked,' }} disabled>
```
Comment (sergei-maertens) [inline src/openzaak/templates/snippets/forms.html]:
```suggestion
  <input name="{{ html_name }}" id="{{ id }}" type="checkbox" class="custom-control-input" {{ raw_value|yesno:'checked,' }} disabled>
```

File: src/openzaak/config/templates/config/config_internal.html (hunk 8 above)
Comment (annashamray) [reply src/openzaak/config/templates/config/config_internal.html]: tooltip is added

File: src/openzaak/config/templates/config/config_internal.html (hunk 9 above)
Comment (annashamray) [reply src/openzaak/config/templates/config/config_internal.html]: tooltip is added

File: src/openzaak/config/templates/config/config_detail.html (hunk 6 above)
Comment (annashamray) [reply src/openzaak/config/templates/config/config_detail.html]: tooltip is added

File: src/openzaak/config/templates/config/config_detail.html (hunk 7 above)
Comment (annashamray) [reply src/openzaak/config/templates/config/config_detail.html]: tooltip is added

File: src/openzaak/utils/middleware.py [hunk 12]
```diff
@@ -66,3 +70,31 @@ def _get_version(self, path: str) -> Optional[str]:
...
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        return response
+
+    @staticmethod
+    def get_component_type(request):
+        url = request.path
+        # All component names match the first parts of urls
+        component = url.strip("/").split("/")[0]
```
[the remark this answers is not in the document]
Comment (annashamray) [reply src/openzaak/utils/middleware.py]: Thanks for catching it! Fixed

**Pull request merged**